src/client/client.internal.ts

```diff
@@ -0,0 +1,75 @@
+import { Config } from '../config';
+import {
+  profileFileNotFoundError,
+  profileNotInstalledError,
+  unconfiguredProviderInPriorityError,
+} from '../internal/errors.helpers';
+import { SuperJson } from '../internal/superjson';
+import { Events } from '../lib/events';
+import { IFileSystem } from '../lib/io';
+import { ILogger } from '../lib/logger/logger';
+import { SuperCache } from './cache';
+import { Profile, ProfileConfiguration } from './profile';
+import { IBoundProfileProvider } from './profile-provider';
+
+export class InternalClient {
+  constructor(
+    private readonly events: Events,
+    private readonly superJson: SuperJson,
+    private readonly config: Config,
+    private readonly fileSystem: IFileSystem,
+    private readonly boundProfileProviderCache: SuperCache<{
+      provider: IBoundProfileProvider;
+      expiresAt: number;
+    }>,
+    private readonly logger?: ILogger
+  ) {}
+
+  async getProfile(profileId: string): Promise<Profile> {
+    const profileConfiguration = await this.getProfileConfiguration(profileId);
+
+    return new Profile(
+      profileConfiguration,
+      this.events,
+      this.superJson,
+      this.config,
+      this.fileSystem,
+      this.boundProfileProviderCache,
+      this.logger
+    );
+  }
+
+  public async getProfileConfiguration(
```

Comment:
nitpick: `public` is explicitly used here but only implicitly used above in `getProfile`.